common/views/components/UserInitiatedDialog/UserInitiatedDialog.js

```diff
@@ -0,0 +1,247 @@
+  'aria-controls': 'user-initiated-dialog-window',
+  tabIndex: props.isActive ? '-1' : '0',
+  as: 'button',
+  v: { size: 'm', properties: ['padding-top', 'padding-bottom'] },
+  h: { size: 'l', properties: ['padding-left', 'padding-right'] },
+  className: classNames({
+    [font('hnm', 5)]: true,
+    'plain-button line-height-1 flex-inline flex--v-center bg-hover-purple font-purple font-hover-white': true,
+  }),
+}))`
+  position: fixed;
+  transform: ${props =>
+    props.isActive || !props.shouldStartAnimation
+      ? 'translateY(10px)'
+      : 'translateY(0px)'};
+  bottom: 20px;
+  left: 20px;
+  z-index: 1;
+  background: ${props => props.theme.colors.white};
+  opacity: ${props => (props.isActive || !props.shouldStartAnimation ? 0 : 1)};
+  transition: opacity 500ms ease, filter 500ms ease, transform 500ms ease;
+  transition-delay: ${props => (props.isActive ? '0ms' : '500ms')};
+  border-radius: 9999px;
+  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.3);
+
+  &:hover,
+  &:focus {
+    outline: 0;
+    border: 0;
+
+    .icon__shape {
+      fill: ${props => props.theme.colors.white};
+    }
+  }
+`;
+
+const UserInitiatedDialogWindow = styled(Space).attrs(props => ({
+  'aria-modal': true,
+  id: 'user-initiated-dialog-window',
+  v: { size: 'l', properties: ['padding-top', 'padding-bottom'] },
+  h: { size: 'l', properties: ['padding-left', 'padding-right'] },
+  className: classNames({
+    'bg-white font-purple': true,
+  }),
+}))`
+  border-radius: 20px 0 20px 0;
+  box-shadow: 0 2px 60px 0 rgba(0, 0, 0, 0.7);
+  opacity: ${props => (props.isActive ? 1 : 0)};
+  pointer-events: ${props => (props.isActive ? 'all' : 'none')};
+  transform: ${props =>
+    props.isActive ? 'translateY(0px)' : 'translateY(10px)'};
+  transition: opacity 500ms ease, transform 500ms ease;
+  transition-delay: ${props => (props.isActive ? '500ms' : '0ms')};
+  position: fixed;
+  bottom: 20px;
+  left: 20px;
+  right: 20px;
+  max-width: 500px;
+  z-index: 1;
+`;
+
+const UserInitiatedDialogClose = styled.button.attrs({
+  className: classNames({
+    'absolute plain-button no-margin no-padding flex flex--v-center flex--h-center': true,
+  }),
+})`
+  top: 10px;
+  right: 10px;
+`;
+
+const UserInitiatedDialogCTA = styled(Space).attrs({
+  as: 'a',
+  v: { size: 'm', properties: ['padding-top', 'padding-bottom'] },
+  h: { size: 'm', properties: ['padding-left', 'padding-right'] },
+  className: classNames({
+    [font('hnm', 5)]: true,
+    'bg-purple font-white font-hover-purple bg-hover-white rounded-corners': true,
+  }),
+})`
+  transition: all 500ms ease;
+  border: 2px solid transparent;
+  text-decoration: none;
+
+  &:hover,
+  &:focus {
+    outline: 0;
+    border-color: ${props => props.theme.colors.purple};
+  }
+`;
+
+type Props = {|
+  openButtonText: string,
+  children: Node,
+  cta: Link,
+|};
+
+const UserInitiatedDialog = ({ children, openButtonText, cta }: Props) => {
+  const [shouldRender, setShouldRender] = useState(false);
```

Comment:
So by default this won't render?

Reply:
Correct. Not until `useEffect` runs, so it won't render if you don't have JS.